src/app/collective-rewards/components/CallToActionCard/CallToActionCard.tsx

```diff
@@ -1,18 +1,87 @@
 import { CommonComponentProps } from '@/components/commonProps'
 import { cn } from '@/lib/utils'
-import { FC, ReactNode } from 'react'
+import { ChevronDownIcon } from 'lucide-react'
+import { motion, AnimatePresence } from 'motion/react'
+import { createContext, FC, ReactNode, useContext, useState } from 'react'
+
+interface CallToActionCardContext {
+  isOpen: boolean
+  toggle: () => void
+}
+
+const CallToActionCardContext = createContext<CallToActionCardContext | null>(null)
 
 interface CallToActionCardProps extends CommonComponentProps {
-  banner?: ReactNode
-  title: ReactNode
+  defaultOpen?: boolean
+  children: ReactNode
+  className?: string
 }
 
-export const CallToActionCard: FC<CallToActionCardProps> = ({ banner, title, children, className = '' }) => {
+export const CallToActionCard = ({ defaultOpen = true, children, className = '' }: CallToActionCardProps) => {
+  const [isOpen, setIsOpen] = useState(defaultOpen)
+  const toggle = () => setIsOpen(prev => !prev)
+
   return (
-    <div data-testid="CallToActionCard" className={cn('flex flex-col flex-1 rounded-sm', className)}>
-      {banner}
-      {title}
-      {children}
+    <CallToActionCardContext.Provider value={{ isOpen, toggle }}>
+      <div className={cn('flex flex-col flex-1 w-full', className)}>{children}</div>
+    </CallToActionCardContext.Provider>
+  )
+}
+
+interface ChildProps {
+  children: ReactNode
+  className?: string
+}
+
+const Banner = ({ children }: { children: ReactNode }) => <div className="hidden md:block">{children}</div>
+
+const Content = ({ children, className }: ChildProps) => <div className={className}>{children}</div>
+
+const Collapsible = ({ children, className }: ChildProps) => {
+  const ctx = useContext(CallToActionCardContext)
+  if (!ctx) throw new Error('CallToActionCard.Collapsible must be used within CallToActionCard')
+
+  return (
+    <AnimatePresence>
+      {ctx.isOpen && (
+        <motion.div
+          initial={{ opacity: 0, height: 0 }}
+          animate={{ opacity: 1, height: 'auto' }}
+          exit={{ opacity: 0, height: 0 }}
+          transition={{
+            duration: 0.3,
+            ease: 'easeInOut',
+            height: { duration: 0.3 },
+            opacity: { duration: 0.2 },
+          }}
+          style={{ overflow: 'hidden' }}
+          className={className}
+        >
+          {children}
+        </motion.div>
+      )}
+    </AnimatePresence>
+  )
+}
+
+const Toggle = () => {
+  const ctx = useContext(CallToActionCardContext)
+  if (!ctx) throw new Error('CallToActionCard.Toggle must be used within CallToActionCard')
```

Comment:
Does it make sense to define a `useCallToActionCardContext` to avoid repetition?